WordPress/src/main/java/org/wordpress/android/ui/media/MediaPreviewActivity.java

```diff
@@ -0,0 +1,262 @@
+
+        setContentView(R.layout.media_preview_activity);
+        mImageView = (ImageView) findViewById(R.id.image_preview);
+        mVideoView = (VideoView) findViewById(R.id.video_preview);
+
+        if (savedInstanceState != null) {
+            mContentUri = savedInstanceState.getString(ARG_MEDIA_CONTENT_URI);
+            mMediaId = savedInstanceState.getInt(ARG_MEDIA_LOCAL_ID);
+            mIsVideo = savedInstanceState.getBoolean(ARG_IS_VIDEO);
+        } else {
+            mContentUri = getIntent().getStringExtra(ARG_MEDIA_CONTENT_URI);
+            mMediaId = getIntent().getIntExtra(ARG_MEDIA_LOCAL_ID, 0);
+            mIsVideo = getIntent().getBooleanExtra(ARG_IS_VIDEO, false);
+        }
+
+        mImageView.setVisibility(mIsVideo ?  View.GONE : View.VISIBLE);
+        mVideoView.setVisibility(mIsVideo ? View.VISIBLE : View.GONE);
+
+        String mediaUri;
+        if (!TextUtils.isEmpty(mContentUri)) {
+            mediaUri = mContentUri;
+        } else if (mMediaId != 0) {
+            MediaModel media = mMediaStore.getMediaWithLocalId(mMediaId);
+            if (media == null) {
+                delayedFinish(true);
+                return;
+            }
+            mediaUri = media.getUrl();
+        } else {
+            delayedFinish(true);
+            return;
+        }
+
+        if (mIsVideo) {
+            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+            playVideo(mediaUri);
+        } else {
+            loadImage(mediaUri);
+        }
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putString(ARG_MEDIA_CONTENT_URI, mContentUri);
+        outState.putInt(ARG_MEDIA_LOCAL_ID, mMediaId);
+        outState.putBoolean(ARG_IS_VIDEO, mIsVideo);
+    }
+
+    private void delayedFinish(boolean showError) {
+        if (showError) {
+            ToastUtils.showToast(this, R.string.error_media_not_found);
+        }
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                finish();
+            }
+        }, 1500);
+    }
+
+    private void showProgress(boolean show) {
+        findViewById(R.id.progress).setVisibility(show ? View.VISIBLE : View.GONE);
+    }
+
+    /*
+     * loads and displays a remote or local image
+     */
+    private void loadImage(@NonNull String mediaUri) {
+        int width = DisplayUtils.getDisplayPixelWidth(this);
+        int height = DisplayUtils.getDisplayPixelHeight(this);
+
+        if (mediaUri.startsWith("http")) {
+            showProgress(true);
+            String imageUrl = PhotonUtils.getPhotonImageUrl(mediaUri, width, height);
+            mImageLoader.get(imageUrl, new ImageLoader.ImageListener() {
+                @Override
+                public void onResponse(ImageLoader.ImageContainer response, boolean isImmediate) {
+                    if (response.getBitmap() != null) {
```

Comment:
What will happen in the unlikely event that `getBitmap` returns `null` here?
Should we dismiss the progress bar and show the error?